File: hsg9-tinh/2023-2024-khanhhoa/demgao/demgao.cpp
#include <iostream>

#define input_file "demgao.inp"
#define output_file "demgao.out"

using namespace std;

typedef long long int lli;

lli a, b, k;
lli count_multiples;


void input()
{
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);

    freopen(input_file, "r", stdin);

    cin >> a >> b >> k;
}


void process()
{
    // Số lượng bội của k trong phạm vi [1..b]
    lli count_b_k = b / k;

    // Số lượng bội của k trong phạm vi [1..a - 1]
    lli count_a_1_k = (a - 1) / k; 

    // Số lượng bội của k trong phạm vi [a..b]
    count_multiples = count_b_k - count_a_1_k;
}


void output()
{
    freopen(output_file, "w", stdout);

    cout << count_multiples;
}


int main()
{
    input();
    process();
    output();

    return 0;
}

File: hsg9-tinh/2023-2024-khanhhoa/demgao/demgao.py
import os
import sys

input_file = os.path.join(sys.path[0], 'demgao.inp')
output_file = os.path.join(sys.path[0], 'demgao.out')

a = b = k = 0
count_multiples = 0


def input():
    global a, b, k

    with open(input_file, 'r') as f:
        a, b, k = map(int, f.readline().split())


def process():
    global a, b, k, count_multiples

    # Số lượng bội của k trong phạm vi [1..b]
    count_b_k = b // k

    # Số lượng bội của k trong phạm vi [1..a - 1]
    count_a_1_k = (a - 1) // k 

    # Số lượng bội của k trong phạm vi [a..b]
    count_multiples = count_b_k - count_a_1_k 


def output():
    global count_multiples

    with open(output_file, 'w') as f:
        f.write(str(count_multiples))


if __name__ == '__main__':
    input()
    process()
    output()